docs/tutorials/web/performance-diagnosis.md

```diff
@@ -0,0 +1,197 @@
+
+### Enable multi-threading
+
+Always enable multi-threading if the environment supports it. Multi-threading can significantly improve the performance of your application by utilizing multiple CPU cores.
+
+This feature is enabled by default in ONNX Runtime Web, however it only works when `crossOriginIsolated` mode is enabled. See [https://web.dev/cross-origin-isolation-guide/](https://web.dev/cross-origin-isolation-guide/) for more info.
+
+You can also use flag `ort.env.wasm.numThreads` to set the number of threads to be used.
+
+```js
+// Set the number of threads to 4
+ort.env.wasm.numThreads = 4;
+
+// Disable multi-threading
+ort.env.wasm.numThreads = 1;
+
+// Let ONNX Runtime Web decide the number of threads to use
+ort.env.wasm.numThreads = 0;
+```
+
+See [API reference: env.wasm.numThreads](https://onnxruntime.ai/docs/api/js/interfaces/Env.WebAssemblyFlags.html#numThreads) for more details.
+
+### Enable SIMD
+
+Always enable SIMD if it's supported. SIMD (Single Instruction, Multiple Data) is a set of instructions that perform the same operation on multiple data points simultaneously. This can significantly improve the performance of your application.
+
+This feature is enabled by default in ONNX Runtime Web, unless you explicitly disable it by setting `ort.env.wasm.simd = false`.
+
+See [API reference: env.wasm.simd](https://onnxruntime.ai/docs/api/js/interfaces/Env.WebAssemblyFlags.html#simd) for more details.
+
+### Prefer uint8 quantized models
+
+If you are using a quantized model, prefer uint8 quantized models. Avoid float16 models if possible, as float16 is not natively supported by CPU and it is going to be slow.
+
+### Enable Proxy Worker
+
+Proxy worker is a feature that allows ONNX Runtime Web to offload the heavy computation to a separate Web Worker. Using the proxy worker cannot improve the performance of the model, but it can improve the responsiveness of the UI to improve the user experience.
+
+If you didn't import ONNX Runtime Web in a Web Worker, and the model takes a while to inference, it is recommended to enable the proxy worker.
+
+```js
+// Enable proxy worker
+ort.env.wasm.proxy = true;
+```
+
+See [API reference: env.wasm.proxy](https://onnxruntime.ai/docs/api/js/interfaces/Env.WebAssemblyFlags.html#proxy) for more details.
+
+## WebGPU tips
+
+If you are using the WebGPU execution provider, you can use the following tips to improve the performance of your application:
+
+### Use capture and replay if possible
+
+See [Capture and replay](ep-webgpu.md#capture-and-replay) for feature introduction.
+
+It is always recommended to enable capture and replay feature, unless you need to feed input data with dynamic shape (eg. transformer based decoder model). Even with static shape input, this feature does not always work for all models. You can try it out and see if it works for your model. If it doesn't work, the model initialization will fail, and you can disable this feature for this model.
```

Comment:
I think we don't want to say 'it is always recommended' ... maybe something like 'if your model has static shapes and runs long sequences of ops on webgpu' ...
Not sure if we stressed this feature enough, never used it for benchmarks.

Reply:
My understanding is that, if the model inference time is small enough, this feature can reduce a lot on the E2E inference time.

In my understanding, it always worth a try, nothing to loss.